test/utils/XmlHelperSpec.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+
+import scala.xml.Node
+import scala.xml.Utility.trim
+
+class XmlHelperSpec extends AnyFreeSpec with Matchers {
+
+  object TestXmlHelper$ extends XmlHelper
+
+  "Valid XML" - {
+
+    "without a namespace that is unaltered" in {
+      val input = <root>
+        <child>test</child>
+      </root>
+
+      TestXmlHelper$.stripNamespaceFromRoot(input) mustEqual input
+    }
+
+    "with a namespace that is altered" in {
+      val inputs = Seq(
+        <root xmlns:xsi="test">
+          <child>test</child>
+        </root>,
+        <root xmlns:xsd="test">
+          <child>test</child>
+        </root>,
+        <root xmlns:xsd="test" xmlns:xsi="test">
+          <child>test</child>
+        </root>
+      )
+
```

Comment:
Is it worth adding an attribute to the child element to ensure the recursion is working as expected?

Reply:
Discussed this in mob, but to summarise - yes and no. It's a good idea if we _also_ have a root element with a namespace on it, as that'll be an explicit verification of the recursive algorithm, so I'll add that! However, not in the case of if only the child element has it, in the large message trees we get, I don't think we want to traverse every single message we get - I don't think it's likely we'll get a message that has a namespace on a child only.